### PR TITLE
Add client credentials authentication

### LINKED ISSUE #2026: Using Azure Storage Blob SDK
### Problem

HI Team,

Need to create a file inside Azure Storage. To achieve this use case need to use wso2 Azure Storage connector. However the current authentication method provided in the Azure Storage connector is only with account name and account key, but in this use case, need to connect using client id + client secret + tenant id.

**Ex:**

```
export AZURE_CLIENT_ID="31500***bbeef"

export AZURE_TENANT_ID="3197****612d"

export AZURE_CLIENT_SECRET="jrj8***QzIdbW"

export STORAGE_ACCOUNT_ENDPOINT="https://****.blob.core.windows.net/"
```

### Solution

Non

### Affected Component

MI

### Version

_No response_

### Implementation

_No response_

### Related Issues

_No response_

### Suggested Labels

connector

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Add client credentials authentication. A sample config is as follows,

```xml
<msazurestorage.init>
    <connectionName>AZURE_CONNECTION</connectionName>
    <connectionType>azureStorage</connectionType>
    <clientId>YOUR_CLIENT_ID</clientId>
    <clientSecret>YOUR_CLIENT_SECRET</clientSecret>
    <tenantId>YOUR_TENANT_ID</tenantId>
    <accountName>STORAGE_ACC_NAME</accountName>
</msazurestorage.init>
```

Fixes: https://github.com/wso2/api-manager/issues/2026

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -197,10 +198,20 @@
...
         </dependency>
+        <dependency>
+            <groupId>com.azure</groupId>
+            <artifactId>azure-identity</artifactId>
+            <version>${azure.identity.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.azure</groupId>
+            <artifactId>azure-core-http-netty</artifactId>
+            <version>1.13.5</version>
```
Comment (GDLMadushanka) [inline pom.xml]: shall we add a property

File: src/main/java/org/wso2/carbon/connector/azure/storage/AzureConfig.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            connectionName = configuration.getConnectionName();
+
+            ConnectionHandler handler = ConnectionHandler.getConnectionHandler();
+            if (!handler.checkIfConnectionExists(connectorName, connectionName)) {
+                AzureStorageConnectionHandler azureStorageConnectionHandler = new AzureStorageConnectionHandler(configuration);
+                handler.createConnection(AzureConstants.CONNECTOR_NAME, connectionName, azureStorageConnectionHandler);
+            } else {
+                AzureStorageConnectionHandler connectionHandler = (AzureStorageConnectionHandler) handler
+                        .getConnection(connectorName, connectionName);
+                if (connectionHandler.getConnectionConfig() != configuration) {
```
Comment (GDLMadushanka) [inline src/main/java/org/wso2/carbon/connector/azure/storage/AzureConfig.java]: do we need reference equality here or object equality ?

Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/azure/storage/AzureConfig.java]: Fixed with 8306278

File: pom.xml (hunk 1 above)
Comment (SanojPunchihewa) [reply pom.xml]: Fixed with 8306278

**Review by rosensilva — APPROVED** [review]

**Pull request merged**